qpdf/fix-qdf.cc

```diff
@@ -380,13 +372,20 @@ realmain(int argc, char* argv[])
     } else if (argc == 2) {
         filename = argv[1];
     }
-    std::string input;
-    if (filename == nullptr) {
+    if (filename.empty()) {
         filename = "standard input";
-        QUtil::binary_stdin();
-        input = QUtil::read_file_into_string(stdin);
     } else {
-        input = QUtil::read_file_into_string(filename);
+#ifdef _WIN32
```

Comment:
I don't want to see any `#ifdef _WIN32` outside of QUtil or code whose specific purpose is encapsulating platform differences. This needs to be encapsulated in a different way. Maybe `QUtil::call_main_from_wmain` can be used. You can see how qpdf.cc does it.

Reply:
Can you give me some more guidance? I'm not sure how to solve this problem. I need some way to pass a wide string to `open`.